### PR TITLE
[LT] Upstream more util functions

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69277
* __->__ #69098

Summary: Add the following utils: helpers, ir_dump_util, and
tensor_util. Some of the util functions may be better organized by
grouping into different files, but we can leave that for later.

Differential Revision: [D32758480](https://our.internmc.facebook.com/intern/diff/D32758480)

### CODE REVIEW

File: torch/csrc/lazy/backend/backend_interface.cpp
Comment (alanwaketan) [inline torch/csrc/lazy/backend/backend_interface.cpp:6]: Oops, you might need to rebase after I land: https://github.com/pytorch/pytorch/pull/69064.

Comment (desertfire) [reply torch/csrc/lazy/backend/backend_interface.cpp:6]: Yes, I noticed that. I will merge after you land yours. I also realized I need to add TORCH_API in those header files.

Comment (alanwaketan) [reply torch/csrc/lazy/backend/backend_interface.cpp:6]: Yeah, most of those are not inline.

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32758480).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32758480).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32758480).

File: torch/csrc/lazy/core/helpers.h
Comment (alanwaketan) [inline torch/csrc/lazy/core/helpers.h:1]: Sounds like we could consolidate the file and util.h. Can you leave a TODO as a follow up?

File: torch/csrc/lazy/core/tensor_util.h
Comment (alanwaketan) [inline torch/csrc/lazy/core/tensor_util.h:31]: I don't think you need `static inline` in functions defined in headers. Is it coding style?

File: torch/csrc/lazy/core/tensor_util.h
Comment (alanwaketan) [inline torch/csrc/lazy/core/tensor_util.h:36]: I don't think you need static inline in functions defined in headers. Is it coding style?

File: torch/csrc/lazy/core/tensor_util.h
Comment (alanwaketan) [inline torch/csrc/lazy/core/tensor_util.h:49]: I don't think you need static inline in functions defined in headers. Is it coding style?

**Review by alanwaketan — APPROVED** [review]
LGTM in general. Can you please address some of the comments before landing?

File: torch/csrc/lazy/core/helpers.h
Comment (desertfire) [reply torch/csrc/lazy/core/helpers.h:1]: Agree. I have mentioned it in the PR description, but let me add a TODO to make it explicit.

File: torch/csrc/lazy/core/tensor_util.h
Comment (desertfire) [reply torch/csrc/lazy/core/tensor_util.h:31]: I have seen both `static inline` and `inline` get used in core. There is actually a lint warning for `static`, so this might be a good time to remove it.

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32758480).

File: torch/csrc/lazy/core/tensor_util.h
Comment (alanwaketan) [reply torch/csrc/lazy/core/tensor_util.h]: Just for my education, what are the benefits/effects of using static in non-member functions in headers? For `inline`, functions defined in header should be implicitly `inline` unless I missed something.

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32758480).

**Pull request closed without merging**
